internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -497,24 +511,28 @@ func doesAdvancedReplicationSpecMatchAPI(tfObject map[string]any, apiObject *adm
 	return tfObject["id"] == apiObject.GetId() || (tfObject["id"] == nil && tfObject["zone_name"] == apiObject.GetZoneName())
 }
 
-func flattenAdvancedReplicationSpec(ctx context.Context, apiObject *admin20231115.ReplicationSpec, tfMapObject map[string]any,
-	d *schema.ResourceData, connV2 *admin20231115.APIClient) (map[string]any, error) {
+func flattenAdvancedReplicationSpecOldSDK(ctx context.Context, apiObject *admin20231115.ReplicationSpec, tfMapObject map[string]any,
+	d *schema.ResourceData, connV2 *admin.APIClient) (map[string]any, error) {
 	if apiObject == nil {
 		return nil, nil
 	}
+	var rootDiskSizeGB *float64
+	if v, ok := d.GetOk("disk_size_gb"); ok {
+		rootDiskSizeGB = conversion.Pointer(v.(float64))
+	}
```

Comment:
I believe this was copied from the expander but does not work in the context of a flattener, we would have to pass the value define in the root of the api response

Reply:
this does work because I set it in the schema. But I see that could be hard to follow, let me refactor

Reply:
ohh I see, had missed that

Reply:
have refactored